Scale-test-action various improvements

marseel opened the pull request:
This PR allows us to use upstream kubernetes/perf-tests in scale-test in cilium/cilium

Exmaple run:
https://github.com/cilium/cilium/actions/runs/7410418894/job/20162692138
Related PR in cilium/cilium: https://github.com/cilium/cilium/pull/29859

learnitall reviewed (changes requested):
Great work! Amazing! Very excited to start using upstream. The only suggestion I have is to expand on the commit messages to explain the full context. This will make it easier to come back to if needed. Here's all the context I can think of:

* HTTP etcd endpoint: We disabled the `TestMetrics` bundle in https://github.com/cilium/perf-tests/commit/6f1bf283e22aa4c29d5db2a7b52501b091a0912e because it includes a measurement which tries to scrape etcd off of an HTTPS endpoint and we were running into https://github.com/golang/go/issues/37278. This issue in go has been fixed, but CL2 has not updated to pull the required change in.
* Adding anonymousAuth: While scraping metrics from the Kubernetes API server, prometheus was getting 401 errors. A bearer token was given to prometheus in commit https://github.com/kubernetes/perf-tests/commit/28b644204e30eb2a38a59939cf1a3c4c9f7f2cae, however this issue can also be solved by configuring the API server to allow anonymous authentication. See https://kubernetes.io/docs/reference/access-authn-authz/authentication/#anonymous-requests for more details.

I think we could also use some context for the extra params that you added, which required the newer version of kops.

marseel commented:
> Adding anonymousAuth: While scraping metrics from the Kubernetes API server, prometheus was getting 401 errors. A bearer token was given to prometheus in commit https://github.com/kubernetes/perf-tests/commit/28b644204e30eb2a38a59939cf1a3c4c9f7f2cae, however this issue can also be solved by configuring the API server to allow anonymous authentication. See https://kubernetes.io/docs/reference/access-authn-authz/authentication/#anonymous-requests for more details.

Sure, I have updated it. FYI I tried to upstream this change, but then it fails in kubemark test case as Prometheus is running in one cluster but scraping target in another so using bearerToken doesn't make sense in that case.

> HTTP etcd endpoint: We disabled the TestMetrics bundle in https://github.com/cilium/perf-tests/commit/6f1bf283e22aa4c29d5db2a7b52501b091a0912e because it includes a measurement which tries to scrape etcd off of an HTTPS endpoint and we were running into https://github.com/golang/go/issues/37278. This issue in go has been fixed, but CL2 has not updated to pull the required change in.

Hmmm, I updated commit message, but your explanation is not entirely true. The initial issue was with ssh as you mentioned, not really http/https part. I have updated x/crypto upstream, but then also once it ssh into node it needs to be able to scrape these metrics without https. That's how upstream is doing it and there is no way currently of scraping through https endpoint IIUC.
